### PR TITLE
docs: various documentation updates

### PR DESCRIPTION
**Pull request by OrKoN**
Mostly updating the selector documentation and copying it in all APIs so that it is easy to find. And adding a deprecation notice to isOOPIFFrame method.

### CODE REVIEW

File: docs/api/puppeteer.page.__eval.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ sidebar_label: Page.$$eval
 
 # Page.$$eval() method
 
-This method runs `Array.from(document.querySelectorAll(selector))` within the page and passes the result as the first argument to the `pageFunction`.
+This method returns all elements matching the selector and passes the resulting array as the first argument to the`pageFunction`.
```
Comment (Lightning00Blade) [inline docs/api/puppeteer.page.__eval.md]: Nit:
```suggestion
This method returns all elements matching the selector and passes the resulting array as the first argument to the `pageFunction`.
```

File: docs/api/puppeteer.page.authenticate.md [hunk 2]
```diff
@@ -6,6 +6,12 @@ sidebar_label: Page.authenticate
 
 Provide credentials for `HTTP authentication`.
 
+:::note
```
Comment (Lightning00Blade) [inline docs/api/puppeteer.page.authenticate.md:9]: This is true for CDP as well. Why do we need to mention an implementation detail?

File: docs/api/puppeteer.page.exposefunction.md [hunk 3]
```diff
@@ -8,6 +8,12 @@ The method adds a function called `name` on the page's `window` object. When cal
 
 If the puppeteerFunction returns a `Promise`, it will be awaited.
 
+:::caution
+
+The function will only be installed on the main frame and not child frames.
```
Comment (Lightning00Blade) [inline docs/api/puppeteer.page.exposefunction.md]: Is this correct? We do `Page.addScriptToEvaluateOnNewDocument` and `evaluate` on each current frame

File: packages/puppeteer-core/src/api/Page.ts [hunk 4]
```diff
@@ -1570,15 +1648,14 @@ export abstract class Page extends EventEmitter<PageEvents> {
...
    * - the remote server does not respond or is unreachable.
-   * - the main resource failed to load.
    *
-   * This method will not throw an error when any valid HTTP status code is
-   * returned by the remote server, including 404 "Not Found" and 500 "Internal
-   * Server Error". The status code for such responses can be retrieved by
-   * calling {@link HTTPResponse.status}.
+   * - the main resource failed to load.
    */
   async goto(url: string, options?: GoToOptions): Promise<HTTPResponse | null> {
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/api/Page.ts:1615]: Can you try: It should be 1:1 that way we don't duplicate the comments
```
{@inheritDoc Frame.goto}
```

File: docs/api/puppeteer.page.exposefunction.md (hunk 3 above)
Comment (OrKoN) [reply docs/api/puppeteer.page.exposefunction.md]: Right, probably not completely accurate but addScriptToEvaluateOnNewDocument does not work for OOPIFs, only called for local frames.

File: docs/api/puppeteer.page.authenticate.md (hunk 2 above)
Comment (OrKoN) [reply docs/api/puppeteer.page.authenticate.md:9]: It might affect performance so I think it's worth leaving a note.

**Review by Lightning00Blade — APPROVED** [review]

**Pull request merged**